paying_for_college/tests/test_views.py

```diff
@@ -1,29 +1,150 @@
 import datetime
 import unittest
 
+import mock
 import django
-from django.http import HttpResponse
+from django.test import RequestFactory
+from django.http import HttpResponse, HttpRequest
 from django.test import Client
 from django.core.urlresolvers import reverse
-from paying_for_college.views import Feedback
+from paying_for_college.views import Feedback, EmailLink, school_search_api
+from paying_for_college.views import SchoolRepresentation
+from paying_for_college.models import School
+from paying_for_college.search_indexes import SchoolIndex
+
 client = Client()
 
 
+def setup_view(view, request, *args, **kwargs):
+    """Mimic as_view() returned callable, return view instance instead."""
+    view.request = request
+    view.args = args
+    view.kwargs = kwargs
+    return view
+
+
 class TestViews(django.test.TestCase):
 
-    def test_landing_view(self):
-        response = client.get(reverse('pfc-landing'))
-        self.assertTrue('base_template' in response.context_data.keys())
+    landing_page_views = [
+        'pfc-landing',
+        'pfc-repay',
+        'pfc-choose',
+        'pfc-manage',
+        'pfc-demo',
+    ]
+    POST = HttpRequest()
+    POST.POST = {'school-program': '999999',
+                 'ba': True,
+                 'is_valid': True}
 
-    def test_stand_alone_view(self):
-        response = client.get(reverse('pfc-repay'))
-        self.assertTrue('base_template' in response.context_data.keys())
+    def test_landing_page_views(self):
+        for url_name in self.landing_page_views:
+            response = client.get(reverse(url_name))
+            self.assertTrue('base_template' in response.context_data.keys())
 
     def test_feedback(self):
         response = client.get(reverse('disclosures:pfc-feedback'))
         self.assertTrue(sorted(response.context_data.keys()) ==
                         ['base_template', 'form'])
 
-    def test_comparison(self):
+    def test_disclosure(self):
         response = client.get(reverse('disclosures:worksheet'))
         self.assertTrue('base_template' in response.context.keys())
+        response2 = client.post(reverse('disclosures:worksheet'),
+                                request=self.POST)
+        self.assertTrue('GET' in '%s' % response2)
+
+    def test_technote(self):
+        response = client.get(reverse('disclosures:pfc-technote'))
+        self.assertTrue('base_template' in response.context_data.keys())
+
+
+# http://www.consumerfinance.gov/paying-for-college/compare-financial-aid-and-college-cost/api/email/
+class EmailTest(django.test.TestCase):
+
+    fixtures = ['test_fixture.json']
+    factory = RequestFactory()
+
+    url = reverse('disclosures:email')
+    post_data = {'id': '00470019-e077-4fc3-9dbb-4a595fe976e6',
+                 'email': 'xxxfakexxx@gmail.com'}
```

Comment:
We might not want to use three Xs here (just for optics and politics and all that fun stuff).